ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpStatementTest.kt

```diff
@@ -32,4 +34,22 @@ class HttpStatementTest : ClientLoader() {
             assertEquals("hello", response.receive<String>())
         }
     }
+
+    @Test
+    fun testGZipFromSavedResponse() = clientTests {
+        config {
+            ContentEncoding {
+                gzip()
+            }
+        }
+
+        test { client ->
+            val response = client.get<HttpResponse>("$TEST_SERVER/compression/gzip")
```

Comment:
Why is it guaranteed here that the result of `get` will be always a `SavedHttpResponse` but not just `HttpResponse`?

Reply:
Because we can get the simple `HttpResponse` only when executing `HttpStatement` with `block`. Otherwise, it will be `SavedHttpResponse`